Plugins/SpatialGDK/SpatialGDKEditorToolbar/Source/SpatialGDKEditorToolbar/Private/InteropCodeGenerator/TypeBindingGenerator.cpp

```diff
@@ -182,18 +229,21 @@ void GenerateUnrealToSchemaConversion(FCodeWriter& Writer, const FString& Update
 			TArray<uint8> ValueData;
 			FSpatialMemoryWriter ValueDataWriter(ValueData, PackageMap, UnresolvedObjects);)""");
 
+		bool bCPPDefinedStruct = !Struct->IsA<UUserDefinedStruct>(); // Special case for array of blueprint-defined structs
+
 		if (Struct->StructFlags & STRUCT_NetSerializeNative)
 		{
+			checkf(bCPPDefinedStruct, TEXT("NetSerialize is not supported for blueprint-defined structs."));
```

Comment:
Can this check ever be hit?

Reply:
I'm not sure, but if it ever did the generated code would fail with unclear debugging info, so it seems good practice to leave it in.

Reply:
I disagree. `check`s imply that a check can be hit if called under invalid circumstances. If an invalid circumstance can't occur then it doesn't make sense as to why a `check` would exist.

Reply:
will change